### PR TITLE
Added offline run flag and profile

### LINKED ISSUE #28: Full offline run
### Description of feature

Add the option to run the pipeline fully offline for large batches (to not get rate limited to death)

### PR DESCRIPTION
**Pull request by itrujnara**
Closes #28 

- added `offline_run` option
- added `offline` profile

Note: the options only affect the ortholog fetching, downstream still requires network. FASTA input is not available for offline runs for now.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).

### CODE REVIEW

File: conf/offline.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+params {
+    config_profile_name        = 'Offline'
+    config_profile_description = 'Settings for offline run'
+
+    // Other parameters
+    offline_run     = true
+    local_databases = true
```
Comment (JoseEspinosa) [inline conf/offline.config]: Don't you need also to provide here the paths to the local DBs?

File: subworkflows/local/get_orthologs.nf [hunk 2]
```diff
@@ -28,6 +28,14 @@ workflow GET_ORTHOLOGS {
     ch_versions    = Channel.empty()
     ch_orthogroups = Channel.empty()
 
+    fasta_input = false
+    ch_samplesheet_fasta.ifEmpty {
+        fasta_input = true
+    }
+    if (fasta_input && params.offline_run) {
```
Comment (JoseEspinosa) [inline subworkflows/local/get_orthologs.nf:36]: Do we want to through a warning  here instead making clear that using `fasta` could have an impact on the rate limit if too many files are provided?

File: subworkflows/local/get_orthologs.nf [hunk 3]
```diff
@@ -41,14 +49,19 @@ workflow GET_ORTHOLOGS {
...
+        ch_samplesheet_query,
+        params.offline_run
     )
 
     ch_query = IDENTIFY_SEQ_ONLINE.out.seqinfo.mix(WRITE_SEQINFO.out.seqinfo)
     ch_versions = ch_versions.mix(WRITE_SEQINFO.out.versions)
 
     // Ortholog fetching
 
+    if(params.use_all && params.offline_run) {
```
Comment (JoseEspinosa) [inline subworkflows/local/get_orthologs.nf]: Probably, this should go to `subworkflows/local/utils_nfcore_reportho_pipeline/main.nf
```suggestion
        warning("Both '--use_all' and '--offline_run' parameters have been specified!\nThose databases that can't be run offline will be run online.")
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
I added some suggestions. Not sure if we should also check for any DBs that can not be run offline when the `--offline` parameter is set to also trough a warning

File: conf/offline.config (hunk 1 above)
Comment (itrujnara) [reply conf/offline.config]: I don't think I can. I have no prior knowledge of the file structure on the user's machine.

Comment (JoseEspinosa) [reply conf/offline.config]: As discussed on slack I was referring of creating a small version of the databases for running a test using the `offline` mode

Comment (itrujnara) [reply conf/offline.config]: Done now, see `test_offline.config`

**Review by JoseEspinosa — APPROVED** [review]
Thanks! LGTM 🚀

**Pull request merged**